File: java/search_linear.java
class search_linear {
    public static void main(String args[]) {
        int a[] = {2, 0, 4, 5, 1, 3, 6, 7, 8, 9};
        int key = 8;
        int res = -1;

        for (int i = 0; i < a.length; i++) {
            if (a[i] == key) {
                res = i;
                break;
            }
        }

        if (res == -1) {
            System.out.println("Element not found");
        } else {
            System.out.println("Element found at idx: " + res);
        }
    }
}

File: cpp/search_linear.cpp
#include <iostream>

using namespace std;

int compare(int x, int y) {
    return x == y;
}

int search_linear(int *a, int length, int key) {
    int i;
    for (i = 0; i < length; i++) {
        int res = compare(a[i], key);
        if (res == 1) {
            return i;
        }
    }
    return -1;
}

int main() {
    int a [] = {2, 4, 3, 6, 9, 7, 8, 1, 5, 0};
    int length = sizeof(a)/sizeof(a[0]);
    int key = 1;
    int res = search_linear(a, length, key);
    if (res == -1) {
        cout << "Element not found" << "\n";
    } else {
        cout << "Element found at idx: " << res << "\n";
    }
    return 0;
}
